aiida/transports/transport.py

```diff
@@ -41,8 +35,7 @@ def validate_positive_number(ctx, param, value):  # pylint: disable=unused-argum
         raise BadParameter('{} is not a valid positive number'.format(value))
 
 
-@six.add_metaclass(ABCMeta)
-class Transport(object):
```

Comment:
Same comment about the metaclass as above.

Reply:
Good point. I put it for the `Node` class which used the `AbstractNodeMeta` but not sure if it is actually _needed_ for any of the others. Do you know when using ABC as metaclass is crucial as opposed to just making a class with abstract methods? Will the latter fail if class is not explicitly marked as abstract as well?

Reply:
I think I understand now why nothing complained and everything still works when I just removed it. Subclassing from `ABC` is the same as setting the meta to `ABCMeta`. It is just a shortcut. Having `ABCMeta` as meta class is _necessary_ if you want to mark methods as abstract with `@abstractmethod`. Looking at the code where I removed it though, which is mostly very old code, it seems that the original writers never used the `@abstractmethod` decorators, but just raised `NotImplementedError`. This is not statically checked at launch time, unlike the decorators. Instantiating a sub class with an abstract method will raise, but if you don't override the method of the base class, no-one will notice until you call it.

In conclusion, I think these should for the most part be abstract classes but then their methods should be appropriately made abstract as well. The question is now, do I do this in this PR? Or shall I for now just make them subclass `ABC` directly and leave the fixing of the methods for another time?

Reply:
I see, that makes sense.

For the sake of making it a bit easier to track the change, maybe it's best to just subclass ``ABC`` now, and then make a separate PR with the abstractmethods.

Reply:
Done